client/src/components/MainPage/TaskModal/TagInput.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+import { HOST } from '../../../constants';
+import styled from 'styled-components';
+import useInput from '../../../hooks/useInput';
+import { RiCloseLine } from 'react-icons/ri';
+import { Tag } from 'GlobalType';
+
+axios.defaults.withCredentials = true; // withCredentials 전역 설정
+
+interface Props {
+  setTagIdx: React.Dispatch<React.SetStateAction<number | undefined>>;
+}
+
+const TagInput = (props: Props) => {
+  const [tagList, setTagList] = useState<Tag[]>([]);
+  const [tagInput, onTagInputChange, setTagInput] = useInput('');
```

Comment:
`tagInput` → 컴포넌트 인스턴스인 것처럼 느껴집니다.
```suggestion
  const [tagInputValue, onTagInputChange, setTagInputValue] = useInput('');
```